src/string_bytes.cc

```diff
@@ -72,10 +73,15 @@ class ExternString: public ResourceType {
       ExternString* h_str = new ExternString<ResourceType, TypeName>(isolate,
                                                                      data,
                                                                      length);
-      Local<String> str = String::NewExternal(isolate, h_str);
+      MaybeLocal<String> str = String::NewExternal(isolate, h_str);
       isolate->AdjustAmountOfExternalAllocatedMemory(h_str->byte_length());
 
-      return scope.Escape(str);
+      if (str.IsEmpty()) {
+        delete h_str;
+        return Local<String>();
+      }
```

Comment:
Why would it leak `new_data` on error? `new_data` is allocated by `ExternString::NewOnCopy` then used by`ExternString::New(new_data)` which deallocates the inputted data on error with `delete h_str`.